source/allocator.c

```diff
@@ -46,9 +51,15 @@ static void *s_aligned_malloc(struct aws_allocator *allocator, size_t size) {
      * 8 byte alignment for <= page allocations on 32 bit systems
      *
      * We use PAGE_SIZE as the boundary because we are not aware of any allocations of
-     * this size or greater that are not data buffers
+     * this size or greater that are not data buffers.
+     *
+     * Unless there is a customized alignment size.
      */
-    const size_t alignment = sizeof(void *) * (size > (size_t)PAGE_SIZE ? 8 : 2);
+    size_t alignment = sizeof(void *) * (size > (size_t)PAGE_SIZE ? 8 : 2);
+    if (allocator->impl) {
```

Comment:
allocator will not have impl if its aligned by default?

Reply:
dumb idea, but maybe just let impl be the actual alignment (i.e. just cast impl pointer to size_t and back) instead of allocating another chunk of mem for it?